pkg/metrics/osbclientproxy/osbproxy.go

```diff
@@ -0,0 +1,152 @@
+	glog.V(9).Info("OSBClientProxy UpdateInstance()")
+	response, err := pc.realOSBClient.UpdateInstance(r)
+	pc.updateMetrics(err)
+	return response, err
+}
+
+// DeprovisionInstance implements
+// go-open-service-broker-client/v2/Client.DeprovisionInstance by proxying the
+// method to the underlying implementation and capturing request metrics.
+func (pc proxyclient) DeprovisionInstance(r *osb.DeprovisionRequest) (*osb.DeprovisionResponse, error) {
+	glog.V(9).Info("OSBClientProxy DeprovisionInstance()")
+	response, err := pc.realOSBClient.DeprovisionInstance(r)
+	pc.updateMetrics(err)
+	return response, err
+}
+
+// PollLastOperation implements
+// go-open-service-broker-client/v2/Client.PollLastOperation by proxying the
+// method to the underlying implementation and capturing request metrics.
+func (pc proxyclient) PollLastOperation(r *osb.LastOperationRequest) (*osb.LastOperationResponse, error) {
+	glog.V(9).Info("OSBClientProxy PollLastOperation()")
+	response, err := pc.realOSBClient.PollLastOperation(r)
+	pc.updateMetrics(err)
+	return response, err
+}
+
+// PollBindingLastOperation implements
+// go-open-service-broker-client/v2/Client.PollBindingLastOperation by proxying
+// the method to the underlying implementation and capturing request metrics.
+func (pc proxyclient) PollBindingLastOperation(r *osb.BindingLastOperationRequest) (*osb.LastOperationResponse, error) {
+	glog.V(9).Info("OSBClientProxy PollBindingLastOperation()")
+	response, err := pc.realOSBClient.PollBindingLastOperation(r)
+	pc.updateMetrics(err)
+	return response, err
+}
+
+// Bind implements go-open-service-broker-client/v2/Client.Bind by proxying the
+// method to the underlying implementation and capturing request metrics.
+func (pc proxyclient) Bind(r *osb.BindRequest) (*osb.BindResponse, error) {
+	glog.V(9).Info("OSBClientProxy Bind().")
+	response, err := pc.realOSBClient.Bind(r)
+	pc.updateMetrics(err)
+	return response, err
+}
+
+// Unbind implements go-open-service-broker-client/v2/Client.Unbind by proxying
+// the method to the underlying implementation and capturing request metrics.
+func (pc proxyclient) Unbind(r *osb.UnbindRequest) (*osb.UnbindResponse, error) {
+	glog.V(9).Info("OSBClientProxy Unbind()")
+	response, err := pc.realOSBClient.Unbind(r)
+	pc.updateMetrics(err)
+	return response, err
+}
+
+// GetBinding implements go-open-service-broker-client/v2/Client.GetBinding by
+// proxying the method to the underlying implementation and capturing request
+// metrics.
+func (pc proxyclient) GetBinding(r *osb.GetBindingRequest) (*osb.GetBindingResponse, error) {
+	glog.V(9).Info("OSBClientProxy GetBinding()")
+	response, err := pc.realOSBClient.GetBinding(r)
+	pc.updateMetrics(err)
+	return response, err
+}
+
+// updateMetrics bumps the request count metric for the specific broker and
+// status
+func (pc proxyclient) updateMetrics(err error) {
+	if err == nil {
+		metrics.OSBRequestCount.WithLabelValues(pc.brokerName, "200").Inc()
```

Comment:
Could we use 2xx, 3xx, 4xx, and 5xx instead of 200, 300, 400, and 500 so that we are making it clear that it was not necessarily a 200 OK response?

Reply:
Yeah, absolutely, I'll update.